### PR TITLE
Sample extensions for different styling methods

### PR DESCRIPTION
**Pull request by aleksfront**
Adding 3 sample extensions for different styling modifications. All of them based on `HelloWorld` sample.

* `styling-sass-sample` extension using [Sass](https://sass-lang.com/)
* `styling-emotion-sample` extension using CSS-in-JS [Emotion](https://emotion.sh/docs/introduction)
* `styling-css-modules-sample` extension using [CSS Modules](https://github.com/css-modules/css-modules) for class name encapsulation

styling samples

### CODE REVIEW

File: styling-css-modules-sample/renderer.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { LensRendererExtension } from "@k8slens/extensions";
+import { ExampleIcon, ExamplePage } from "./page";
+
+export default class ModulesStylingExtension extends LensRendererExtension {
+  clusterPages = [
+    {
+      path: "/css-modules-styling-sample",
```
Comment (ixrock) [inline styling-css-modules-sample/renderer.tsx]: API is changed, now page-registration and page-menu-regisration are splitted.
Working example:
```ts
  clusterPages = [
    {
      id: "example-page-id",
      components: {
        Page: () => <ExamplePage extension={this}/>,
      }
    }
  ]

  clusterPageMenus = [
    {
      id: this.clusterPages[0].id, // must be the same as in page registration to be visible in sidebar
      title: "Hello World",
      components: {
        Icon: ExampleIcon
      }
    }
  ]
```

File: styling-css-modules-sample/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "sourceMap": false,
+    "declaration": false,
+    "strict": false,
+    "noImplicitAny": true,
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "experimentalDecorators": true,
+    "jsx": "react",
+    "plugins": [{ "name": "typescript-plugin-css-modules" }],
```
Comment (ixrock) [inline styling-css-modules-sample/tsconfig.json:17]: Did't know about that field. How does it work? Where the list of available plugins?

File: styling-sass-sample/styles.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.SassStylingExtension {
+  .doodle-style {
+    width: 200px;
+    background-color: cadetblue;
+    padding: calc(var(--padding) * 2);
+
+    // Check if light theme enabled in Lens
+    .theme-light & {
```
Comment (ixrock) [inline styling-sass-sample/styles.scss:10]: probably we should specify top-level theme element more precisely, like `body[data-theme="light"]`

Comment (aleksfront) [reply styling-sass-sample/styles.scss:10]: I'm working on this, changes will be in different PR.

File: styling-css-modules-sample/renderer.tsx (hunk 1 above)
Comment (aleksfront) [reply styling-css-modules-sample/renderer.tsx]: Starting the app with described changes and got
```
Uncaught TypeError: Expected "name" to match "[^\/#\?]+?", but got "@k8slens/styling-css-modules-sample"
    at LensDev.js:402426
    at n.getPageUrl (LensDev.js:521110)
    at LensDev.js:523074
    at Array.map (<anonymous>)
    at PageMenuRegistry.getItems (LensDev.js:523073)
    at Sidebar.render (LensDev.js:636929)
    at allowStateChanges (LensDev.js:334943)
    at LensDev.js:333372
    at trackDerivedFunction (LensDev.js:334726)
    at Reaction../node_modules/mobx/lib/mobx.module.js.Reaction.track (LensDev.js:335760)
```
with app crash. Something with id?

File: styling-css-modules-sample/tsconfig.json (hunk 2 above)
Comment (aleksfront) [reply styling-css-modules-sample/tsconfig.json:17]: Here is the `plugin` field documentation https://www.typescriptlang.org/tsconfig#plugins
And here is `typescript-plugin-css-modules` page https://github.com/mrmckeb/typescript-plugin-css-modules

File: styling-css-modules-sample/renderer.tsx (hunk 1 above)
Comment (ixrock) [reply styling-css-modules-sample/renderer.tsx]: Upgrade to:
```
"@k8slens/extensions": "^4.0.0-beta.1",
```

Comment (aleksfront) [reply styling-css-modules-sample/renderer.tsx]: Adding a `url` doesn't help either:
```
clusterPages = [
    {
      id: "extension-example",
      components: {
        Page: () => <ExamplePage extension={this}/>,
      }
    }
  ]

  clusterPageMenus = [
    {
      id: this.clusterPages[0].id, // must be the same as in page registration to be visible in sidebar
      url: `/${this.clusterPages[0].id}`,
      title: "Hello World",
      components: {
        Icon: ExampleIcon
      }
    }
  ]
```

Comment (ixrock) [reply styling-css-modules-sample/renderer.tsx]: Ok, probably better to wait fixes from https://github.com/lensapp/lens/pull/1364

**Comment by stefcameron** [comment]
@aleksfront I had a quick look. It's great that you're providing these samples, thank you! What I don't see is the theme switching use case. How will developers handle that? How will they know when the theme changes? Right now, all I see are different ways to write CSS that uses theme-based CSS variables -- but maybe I missed something since I didn't look closely at everything.

**Comment by aleksfront** [comment]
@ixrock Samples working with latest (beta 3) extension api types. Please re-review.
styling extensions

**Review by ixrock — APPROVED** [review]

**Pull request merged**